os/arch/arm/src/imxrt/imxrt_usbhost.c

```diff
@@ -0,0 +1,343 @@
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name NuttX nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ************************************************************************************/
+
+/************************************************************************************
+ * Included Files
+ ************************************************************************************/
+
+#include <tinyara/config.h>
+
+#include <stdio.h>
+#include <sys/types.h>
+#include <stdint.h>
+#include <stdbool.h>
+#include <sched.h>
+#include <errno.h>
+#include <assert.h>
+#include <debug.h>
+
+#include <tinyara/irq.h>
+#include <tinyara/kthread.h>
+#include <tinyara/usb/usbdev.h>
+#include <tinyara/usb/usbhost.h>
+#include <tinyara/usb/usbdev_trace.h>
+
+#include "up_arch.h"
+#include "chip.h"
+#include "imxrt_gpio.h"
+#include "chip/imxrt_pinmux.h"
+#include "imxrt_ehci.h"
+#include "imxrt_log.h"
+
+#ifdef HAVE_USBHOST
+
+/************************************************************************************
+ * Pre-processor Definitions
+ ************************************************************************************/
+
+#ifndef CONFIG_USBHOST_DEFPRIO
+#define CONFIG_USBHOST_DEFPRIO 50
+#endif
+
+#ifndef CONFIG_USBHOST_STACKSIZE
+#ifdef CONFIG_USBHOST_HUB
+#define CONFIG_USBHOST_STACKSIZE 1536
+#else
+#define CONFIG_USBHOST_STACKSIZE 1024
+#endif
+#endif
+
+#undef uerr
+#undef uinfo
+#if defined(DEBUG_IMX_DMA_DRIVER)
+#define uinfo(format, ...)   printf(format, ##__VA_ARGS__)
+#define uerr(format, ...)   printf(format, ##__VA_ARGS__)
+#else
+#define uinfo(format, ...)
+//#define uerr(format, ...)
```

Comment:
Can we remove this line if not required?

Reply:
I fixed it